SalesforceCommerceCloud/pwa-kit
# Correlation ID Implementation

# Description

This PR demonstrates how to generate a correlation ID to SCAPI requests via headers. This PR made the changes on top of the useQuery SSR as we are looking forwards to adding the correlation ID to our commerce-sdk-react hooks rather than the Retail App project



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Pull the code
- Get the token and put it in the commerce-sdk-react provider, build the package
- Start the test project
- Verify that the request is presented in the headers. You can look for it in the browser console on each request. This is client side. Update, we now send the header `correlation-id` instead of `x-correlation-id`

<img width="1345" alt="image" src="https://user-images.githubusercontent.com/52219283/188014156-0ea5ab7a-2873-429f-8ff8-135c5cebfe3a.png">

- Verify that the correlation id is consoled out in the terminal as such. This was the result of tweaking the server logging to promot out more information using the package `morgan` which we has set up but never got to use it much. See [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/709/files#diff-510a9cd2558f6d7aa53b8557cfcd33b95ad6ff97fcf7c596ac1979021a326d86R88)
![image](https://user-images.githubusercontent.com/52219283/188012073-c62d6ffc-f088-4efc-9e02-e3dec8c77de1.png)


Ideally, we should be able to search the request using the correlation ID in splunk. This should verify that our implementation works, but it is not possible yet. It will be implemented by another team. For now, it should be sufficient on our side to verify we have the correct x-correlation-id attached in the request headers that are sent to SCAPI 

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
